lib/rules/require-properties.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * @fileoverview Package properties must be declared
+ * @author zetlen
+ */
+'use strict';
+const {
+    isPackageJson,
+    extractPackageObjectFromAST
+} = require('../processors/PackageJsonProcessor');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: 'Package properties must be declared',
+            category: 'Best Practices'
+        },
+        schema: [
+            {
+                type: 'array',
+                items: {
+                    type: 'string'
+                }
+            }
+        ]
+    },
+
+    create(context) {
+        return {
+            'Program:exit': node => {
+                if (!isPackageJson(context.getFilename())) {
+                    return;
+                }
+                const sourceCode = context.getSourceCode();
+                const packageRoot = extractPackageObjectFromAST(node);
+                const pkg = JSON.parse(sourceCode.getText(packageRoot));
+                const required = context.options[0] || [];
```

Comment:
Should the default value include "name" and "version", since they are required fields? https://docs.npmjs.com/creating-a-package-json-file.
Then this could be a recommended rule.